boa_engine/src/bytecompiler.rs

```diff
@@ -1652,8 +1682,9 @@ impl<'b> ByteCompiler<'b> {
             }
             // TODO: implement AsyncFunctionDecl
             // TODO: implement GeneratorDecl
```

Comment:
Maybe this TODO is no longer needed, right? and the two ones above should be added as comments for the bottom branch.